### PR TITLE
NCL-4593 Provide ability to execute a groovy script when done aligning

### PR DESCRIPTION
**Pull request by geoand**

### CODE REVIEW

**Comment by rnc** [comment]
Heh, I had very similar thoughts, doing something similar to https://github.com/release-engineering/pom-manipulation-ext/blob/master/core/src/main/java/org/commonjava/maven/ext/core/impl/BaseGroovyManipulator.java#L119

File: analyzer/src/functTest/java/org/jboss/gm/analyzer/alignment/SimpleProjectWithCustomGroovyScriptFunctionalTest.java [hunk 1]
```diff
@@ -0,0 +1,95 @@
...
+import org.junit.Before;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.contrib.java.lang.system.RestoreSystemProperties;
+import org.junit.contrib.java.lang.system.SystemOutRule;
+import org.junit.rules.TemporaryFolder;
+import org.junit.rules.TestRule;
+import org.junit.runner.RunWith;
+
+@RunWith(BMUnitRunner.class)
```
Comment (rnc) [inline analyzer/src/functTest/java/org/jboss/gm/analyzer/alignment/SimpleProjectWithCustomGroovyScriptFunctionalTest.java]: I don't think we need to use Byteman runner if there are no bmrules?

File: analyzer/src/main/java/org/jboss/gm/analyzer/alignment/AlignmentTask.java [hunk 2]
```diff
@@ -527,4 +531,20 @@ private String writeProjectNameIfNeeded() throws IOException {
 
         return newProjectName;
     }
+
+    // for now we simply assume that the script is called gme.groovy and already resides in the project's root directory
+    // TODO allow downloading scripts from a configured URL
```
Comment (rnc) [inline analyzer/src/main/java/org/jboss/gm/analyzer/alignment/AlignmentTask.java]: Agreed, lets get it working then reuse PME FileIO code? ( https://github.com/release-engineering/pom-manipulation-ext/blob/master/core/src/main/java/org/commonjava/maven/ext/core/impl/BaseGroovyManipulator.java#L94 )

File: analyzer/src/functTest/java/org/jboss/gm/analyzer/alignment/SimpleProjectWithCustomGroovyScriptFunctionalTest.java (hunk 1 above)
Comment (geoand) [reply analyzer/src/functTest/java/org/jboss/gm/analyzer/alignment/SimpleProjectWithCustomGroovyScriptFunctionalTest.java]: Certainly, I guess it got left over from copy pasta

File: analyzer/src/main/java/org/jboss/gm/analyzer/alignment/AlignmentTask.java (hunk 2 above)
Comment (geoand) [reply analyzer/src/main/java/org/jboss/gm/analyzer/alignment/AlignmentTask.java]: Makes perfect sense!

File: analyzer/src/functTest/java/org/jboss/gm/analyzer/alignment/SimpleProjectWithCustomGroovyScriptFunctionalTest.java (hunk 1 above)
Comment (geoand) [reply analyzer/src/functTest/java/org/jboss/gm/analyzer/alignment/SimpleProjectWithCustomGroovyScriptFunctionalTest.java]: Fixed

**Comment by rnc** [comment]
I think (again for future) we could consider doing something like https://mrhaki.blogspot.com/2013/11/groovy-goodness-set-script-class-with.html (similar to PME) or possibly a DelegatingScript - to give better stronger typing in the groovy.

**Comment by geoand** [comment]
Agreed! We can (and should) make plenty of improvements to the Groovy support in the future.
My only hope is that a limited number of projects will actually need groovy support in the first place 😉

**Comment by rnc** [comment]
Ok to merge - I think we should discuss internally a potential script using this to verify it works.

**Comment by geoand** [comment]
OK, I'll merge for the time being and let's see on Monday how to move forward

**Pull request merged**